src/Orleans/Streams/Predicates/FilterPredicateWrapperData.cs

```diff
@@ -50,6 +50,9 @@ internal class FilterPredicateWrapperData : IStreamFilterPredicateWrapper
         [NonSerialized]
         private StreamFilterPredicate predicateFunc;
 
+        // constructor used by serializator
+        private FilterPredicateWrapperData() { }
```

Comment:
I don't understand what does it mean: "checking for invariants in constructors."
What invariants ?

Would that change mean that ANY class that we generate a serializer for has to have a parameterless constructor? If not, what is the exact restriction?